rdwatch/core/views/animation.py

```diff
@@ -0,0 +1,211 @@
+import logging
+from collections.abc import Iterable
+
+from celery.result import AsyncResult
+from ninja import Router
+from pydantic import UUID4, BaseModel
+
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import get_object_or_404
+
+from rdwatch.core.models import AnimationModelRunExport, AnimationSiteExport
+from rdwatch.core.tasks.animation_export import (
+    GenerateAnimationSchema,
+    create_animation,
+    create_modelrun_animation_export,
+    create_site_animation_export,
+)
+
+logger = logging.getLogger(__name__)
+
+router = Router()
+
+
+def generate_download_data(
+    exports: Iterable[AnimationSiteExport] | Iterable[AnimationModelRunExport],
+):
+    output = []
+    for export in exports:
+        if export.export_file:  # File exists it can be downloaded
+            output.append(
+                {
+                    'created': export.created,
+                    'completed': True,
+                    'arguments': export.arguments,
+                    'taskId': export.celery_id,
+                }
+            )
+        else:
+            task_id = export.celery_id
+            task = AsyncResult(task_id)
+            celery_data = {}
+            celery_data['state'] = task.state
+            celery_data['status'] = task.status
+            celery_data['info'] = (
+                str(task.info) if isinstance(task.info, RuntimeError) else task.info
+            )
```

Comment:
Didn't think this one through and it breaks the logging for the state for the download.  Changing it to `Exception` instead of `runtimeError` should hopefully catch all errors and stringify them.  The task info is an object that contains the current/total and modes.